### PR TITLE
Add CI with conda dependencies

### PR DESCRIPTION
**Pull request by RiccardoGrieco**
This PR addresses https://github.com/robotology/yarp-devices-forcetorque/issues/32.

The workflow uses dependencies installed via `conda` and builds all the software in the repo that can be built without external SDK (e.g. does not build the AMTI FT platform related software).

### CODE REVIEW

File: .github/workflows/conda-ci.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        mamba-version: "*"
+        channels: conda-forge,defaults
+        channel-priority: true
```
Comment (traversaro) [inline .github/workflows/conda-ci.yml:29]:
```suggestion
        miniforge-variant: Mambaforge
        miniforge-version: latest
```

It is more convenient to just use mambaforge directly, see https://github.com/robotology/robotology-superbuild/pull/840 .

File: .github/workflows/conda-ci.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    - name: Dependencies
+      shell: bash -l {0}
+      run: |
+        # Compilation related dependencies
+        mamba install cmake compilers make ninja pkg-config
+        # Actual dependencies
+        mamba install -c conda-forge -c robotology yarp
```
[the remark this answers is not in the document]
Comment (RiccardoGrieco) [reply .github/workflows/conda-ci.yml]: Thanks @traversaro, I took an old existing version of the file and didn't notice it.

It has been changed in 66e1e57 with addition of `asio`.

[the remark this answers is not in the document]
Comment (traversaro) [reply .github/workflows/conda-ci.yml]: Yes, unfortunatly all those files have a lot of copy&paste, so there is a lot of non-optimal things around.

**Review by traversaro — APPROVED** [review]

**Comment by RiccardoGrieco** [comment]
In [d4e04fd](https://github.com/robotology/yarp-devices-forcetorque/pull/37/commits/d4e04fd6fe593d517b6b2484f7ea369a693eb719) I changed the `unistd.h` to `<threads>` when needed or guarded them only when using UNIX. For the second case (https://github.com/robotology/yarp-devices-forcetorque/pull/37/commits/d4e04fd6fe593d517b6b2484f7ea369a693eb719#diff-170b9b1b12284502eabecce5ed6a6213a56564b88d1359551c9e80a6a3b20775) most likely it was working and therefore not guarded because `MinGW` (which provides that header) was being used.

Merging the PR.

**Pull request merged**